features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -73,6 +89,47 @@ void AT_CellularNetwork::urc_no_carrier()
     }
 }
 
+void AT_CellularNetwork::read_reg_params_and_compare(RegistrationType type)
+{
+    RegistrationStatus reg_status = NotRegistered;
+    int lac = -1, cell_id = -1, act = -1;
+
+    read_reg_params(type, reg_status, lac, cell_id, act);
+
+    if (_at.get_last_error() == NSAPI_ERROR_OK) {
+        tr_debug("stat: %d, lac: %d, cellID: %d, act: %d", reg_status, lac, cell_id, act);
```

Comment:
cb could be checked only once, like... if (_at.get_last_error() == NSAPI_ERROR_OK && _connection_status_cb)...

Reply:
good catch, fixed